### PR TITLE
Introduction of gcc-11 to CI

### PR DESCRIPTION
**Pull request by pwisniewskimobica**
update artifact repo url to fork

clean up

add release13 branch

remove cpp option from matrix

remove matrix.include in workflow

install gcc11, g++11

### CODE REVIEW

**Comment by pwisniewskimobica** [comment]
We want to introduce compilation with gcc-11 in Flang CI.
This PR depends on classic llvm PR's: 
https://github.com/flang-compiler/classic-flang-llvm-project/pull/78
https://github.com/flang-compiler/classic-flang-llvm-project/pull/79
https://github.com/flang-compiler/classic-flang-llvm-project/pull/80
https://github.com/flang-compiler/classic-flang-llvm-project/pull/81

File: .github/workflows/build_flang.yml [hunk 1]
```diff
@@ -57,10 +45,10 @@ jobs:
           cd ../..
           curl -sL https://api.github.com/repos/flang-compiler/classic-flang-llvm-project/actions/workflows/pre-compile_llvm.yml/runs --output runs_llvm.json
           wget --output-document artifacts_llvm `jq -r '.workflow_runs[0].artifacts_url?' runs_llvm.json`
-
+
```
Comment (pawosm-arm) [inline .github/workflows/build_flang.yml:48]: what is that line for?

Comment (pwisniewskimobica) [reply .github/workflows/build_flang.yml]: fixed

**Comment by pwisniewskimobica** [comment]
Just as information: CI job now is failing because it is trying to get an artifact from classic flang llvm which is waiting to be merged.

**Comment by pawosm-arm** [comment]
what's the current status of this?

**Comment by pwisniewskimobica** [comment]
@pawosm-arm Current status is that its done on flang side, but first classic llvm project PR's needs to be merged:
flang-compiler/classic-flang-llvm-project: Pull Request 78
flang-compiler/classic-flang-llvm-project: Pull Request 79
flang-compiler/classic-flang-llvm-project: Pull Request 80
flang-compiler/classic-flang-llvm-project: Pull Request 81

**Comment by kiranchandramohan** [comment]
Please rebase.

**Comment by pwisniewskimobica** [comment]
@kiranchandramohan  rebased
-->fixed

**Comment by pwisniewskimobica** [comment]
CI is failing because classic llvm project PR's needs to be merged
flang-compiler/classic-flang-llvm-project: Pull Request 78
flang-compiler/classic-flang-llvm-project: Pull Request 79
flang-compiler/classic-flang-llvm-project: Pull Request 80
flang-compiler/classic-flang-llvm-project: Pull Request 81

**Pull request merged**